core/src/main/java/org/openedx/core/utils/FileUtil.kt

```diff
@@ -1,18 +1,37 @@
 package org.openedx.core.utils
 
 import android.content.Context
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
 import java.io.File
 
-object FileUtil {
+class FileUtil(val context: Context) {
 
-    fun getExternalAppDir(context: Context): File {
+    fun getExternalAppDir(): File {
         val dir = context.externalCacheDir.toString() + File.separator +
                 context.getString(org.openedx.core.R.string.app_name).replace(Regex("\\s"), "_")
         val file = File(dir)
         file.mkdirs()
         return file
     }
 
+    inline fun < reified T> saveObjectToFile(obj: T, fileName: String = "${T::class.java.simpleName}.json") {
```

Comment:
Did you test this with the obfuscated build?

Reply:
No, thanks for your reminder. I made some changes and it works now.